gdal/frmts/pds/isis3dataset.cpp

```diff
@@ -2798,93 +2396,90 @@ void ISIS3Dataset::BuildLabel()
                                 GetMetadataItem("BLOCK_OFFSET_0_0", "TIFF");
             if( pszOffset )
             {
-                oCore["StartByte"] = 1 + atoi(pszOffset);
+                oCore.Set( "StartByte", 1 + atoi(pszOffset) );
```

Comment:
What do you think about the possibility of adding the capability of using oCore["foo"] = bar ? This is somewhat more familiar with the way JSon is manipulated in Python

Reply:
I will try.

Reply:
Just looked `json_object.h` and don't see any simple solution. 
The 'Set' done now via Delete/Add. In case  `CPLJSONObject["key"] = value` we need to self delete Json object and add new one to the parent Json object. So we need to have some pointer to parent in child. 
I afraid it will be risky to have pointer to parent in child as parent may be deleted when reference counter became 0.  If we increase reference counter of parent in child the dead lock may occured.

Have you any ideas?

Reply:
Hum indeed with the approach of CPLJSONObject this seems tricky to do. I think we should perhaps have started with comparing the pros & cons of your approach and the one I did in ISIS3. But as you did the effort of converting ISIS3 and the result looks reasonable, let's go with your approach

Reply:
Good. I tried to make the wrapper as thin as possible. This dictated such an implementation.